src/Documentation/Advanced-Concepts/StatelessWorker.md

```diff
@@ -0,0 +1,84 @@
+``` csharp
+[StatelessWorker]
+public class MyStatelessWorkerGrain() : IMyStatelessWorkerGrain
+{
+ ...
+}
+```
+
+Making a call to a Stateless Worker grain is the same as to any other grain. The only difference is that in most cases a single grain ID is used, 0 or `Guid.Empty`.
+
+``` csharp
+var worker = GrainFactory.GetGrain<IMyStatelessWorkerGrain>(0);
+await worker.Process(args);
+```
+
+
+This one defines a Stateless Worker grain class with no more than one grain activation per silo. 
+``` csharp
+[StatelessWorker(1)]
+public class MyLonelyWorkerGrain() : ILonelyWorkerGrain
+{
+ ...
+}
+```
+
+## State
+
+The "Stateless" part of "Stateless Worker" does not mean that a Stateless Worker cannot have state and is limited only to executing functional operation.
+Like any other grain, a Stateless Worker grain can load and keep in memory any state it needs.
+It's just because multiple activations of a Stateless Worker grain can be created on the same and different silos of the cluster, there is no easy mechanism to coordinate state held by different activations.
+
+There are several useful patterns that involve Stateless Worker holding state.
+
+### Scaled out hot cache items
+
+For hot cache items that experience high throughput, holding each such item in a Stateless Worker grain makes it 
+a) automatically scale out within a silo and across all silos in the cluster; 
+and b) makes the data always locally available on the silo that received the client request via its client gateway, so that the requests can be answered without an extra network hop to another silo.
+
+
+### Reduce style aggregation
+
+In some scenarios applications need to calculate certain metrics across all grains of a particular type in the cluster, and report the aggregates periodically.
+Examples are reporting number of players per game map, average duration of a VOIP call, etc.
+If every of the many thousands or millions of grains were to report their metrics to a single global aggregator, the aggregator would get immediately overloaded unable to process the flood of reports.
+The alternative approach is to turn this task into a 2 (or more) step reduce style aggregation.
+The first layer of aggregation is done by reporting grain sending their metrics to a Stateless Worker pre-aggregation grain.
+The Orleans runtime will automatically create multiple activations of the Stateless Worker grain with each silo.
+Since all such calls will be process locally with no remote calls or serialization of the messages, the cost of such aggregation will be significantly less than in a remote case.
+Now each of the pre-aggregation Stateless Worker grain activations, independently or in coordination with other local activations,
+can send their aggregated reports to the global final aggregator (or to another reduction layer if necessary) without overloading it.
+
```

Comment:
LGTM overall.
A good idea is to add comment about Reentrant attirbute. As far as I remember its orthogonal, meaning  Stateless Worker can be Reentrant or not re-entrant, just like any other grain. I think its not obvious by itself, and some people may assume  Stateless Worker are always re-entrant by default.